### DISCUSSION TITLE
Grouping build pipelines

### DISCUSSION THREAD

**Discussion by kallisti5** [concourse/concourse#8827] [Ideas]
We have been abusing teams to group build pipelines for quite some time.  This is handy to group sets of builds together in the UI for improved clarity... however it comes with a lot of sysadmin overhead of logging into various teams to apply new pipelines, etc.

"It would be nice" if concourse added some grouping (like teams, but without the authentication overhead)

Here's an example of how we're abusing teams:

![screenshot](https://github.com/concourse/concourse/assets/100110/783fb9ba-d9d7-4eed-97e7-f7b9f3b4b717)

https://ci.haiku-os.org

**Comment by kallisti5** [comment]
Before you ask "why aren't you building each architecture as tasks within a pipeline?".
Each one of those builds an entire operating system's worth of stuff :-)

**Reply by marco-m-pix4d** [reply]
Hello,

more or less random comments:

It seems to me that the CI of Concourse itself (https://ci.concourse-ci.org/) is using teams for something very similar to what you are actually doing.

Also if groups existed, you would still need to specify the group name on the command-line, so for one aspect it would be very similar to using `-t team-name`. The only difference is that each N days one has to fly login.

With groups, there would be also the problem of how one could specify the default group for a given team, I think.

I had a look at some of your pipelines, they seem to be made of 1 job. So one possibility is to have, in the same pipeline, one job per operating system, maybe finding a way to logically grouping together a few OSes per pipeline.

Also https://concourse-ci.org/instanced-pipelines.html ("instanced pipelines") could be useful.

Last point, have a look at https://concourse-ci.org/pipelines.html#schema.pipeline.groups
